File: 24_C#Fundamentals/_03_Arrays/_02_Exercise/_05_TopIntegers/_05_TopIntegers.cs
using System;

public class _01_HelloSoftUni
{
    public static void Main(string[] args)
    {
        // Read input values
        int[] numbersArray = Console.ReadLine().Split().Select(int.Parse).ToArray();

        // Add variable
        List<int> topIntegers = new List<int>();

        // Determine the top integers
        for (int i = 0; i < numbersArray.Length; i++)
        {
            bool isBigger = true;

            for (int j = i + 1; j < numbersArray.Length; j++)
            {
                if (numbersArray[i] <= numbersArray[j])
                {
                    isBigger = false;
                    break;
                }
            }

            if (isBigger)
            {
                topIntegers.Add(numbersArray[i]);
            }
        }

        // Print result
        Console.WriteLine(string.Join(" ", topIntegers);
    }
}

File: 02_ProgrammingFundamentals/_03_Arrays/_02_Exercise/_05_TopIntegers.java
package _02_ProgrammingFundamentals._03_Arrays._02_Exercise;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class _05_TopIntegers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int[] numbersArray = Arrays.stream(scan.nextLine().split("\\s+")).mapToInt(Integer::parseInt).toArray();

        // Add variable
        List<Integer> topIntegers = new ArrayList<>();

        // Determine the top integers
        for (int i = 0; i < numbersArray.length; i++) {
            boolean isBigger = true;

            for (int j = i + 1; j < numbersArray.length; j++) {
                if (numbersArray[i] <= numbersArray[j]) {
                    isBigger = false;
                    break;
                }
            }

            if (isBigger) {
                topIntegers.add(numbersArray[i]);
            }
        }

        // Print result
        topIntegers.forEach(e -> System.out.print(e + " "));

        scan.close();
    }
}
